rust/rbac-registration/src/cardano/cip509/utils/cip134_uri_set.rs

```diff
@@ -185,13 +202,53 @@ impl Cip0134UriSet {
                 },
                 C509Cert::C509Certificate(_) => {
                     if let Some(uris) = metadata.certificate_uris.c_uris().get(&index) {
+                        uris.iter().for_each(|v| {
+                            taken_uris.remove(v);
+                        });
                         c_uris.insert(index, uris.clone());
                     }
                 },
             }
         }
 
-        Self(Arc::new(Cip0134UriSetInner { x_uris, c_uris }))
+        Self(Arc::new(Cip0134UriSetInner {
+            x_uris,
+            c_uris,
+            taken_uris,
+        }))
+    }
+
+    /// Return the updated URIs set where the provided URIs were taken by other
+    /// registration chains.
+    ///
+    /// Updates the current URI set by removing the taken URIs from it.
```

Comment:
I think the wording is slightly confusing: this code doesn't remove URIs from the set, it *adds* (marks?) such URIs as taken.